### PR TITLE
adding new component to power adding to catalog with medisearch

### PR DESCRIPTION
**Pull request by billoran8**
this code review contains the necessary file changes for having the 'add to catalog' button be powered by treatments located by medispan's search functionality.

# FULLSCREEN:
Screenshot 2024-09-26 at 6 44 45 PM
Screenshot 2024-09-26 at 6 44 56 PM
Screenshot 2024-09-26 at 6 45 00 PM

# MOBILE:
Screenshot 2024-09-26 at 6 45 13 PM
Screenshot 2024-09-26 at 6 45 16 PM
Screenshot 2024-09-26 at 6 45 33 PM
Screenshot 2024-09-26 at 6 45 36 PM

### CODE REVIEW

File: apps/app/src/views/components/TreatmentOptionSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import { Box, ChakraProvider, InputGroup, InputRightElement, Spinner } from '@chakra-ui/react';
```
Comment (billoran8) [inline apps/app/src/views/components/TreatmentOptionSearch.tsx:1]: first, here's a component that has a searchable dropdown backed by the treatmentOptions search (powered by medidata) that matches the UI elsewhere

File: apps/app/src/views/components/TreatmentOptionSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+}: TreatmentOptionSearchProps) => {
+  const client = usePhoton();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+
+  const [treatmentOptions, setTreatmentOptions] =
+    useState<Array<{ value: string; label: string }>>();
+
+  const clinicalClient = useApolloClient(client?.clinicalClient);
```
Comment (billoran8) [inline apps/app/src/views/components/TreatmentOptionSearch.tsx:43]: is there a more elegant way to reference the correct apollo client? the default one points to the lambdas; the medisearch is on the clinical api.

File: apps/app/src/views/components/TreatmentOptionSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    if (searchTermDebounce) {
+      fetchData();
+    }
+  }, [searchTermDebounce, clinicalClient]);
+
+  const handleSearchTermChange = (searchTerm: string) => {
+    setSearchTerm(searchTerm);
+  };
+
+  const handleSelectedTreatmentChange = (selectedTreatmentId: string) => {
```
Comment (billoran8) [inline apps/app/src/views/components/TreatmentOptionSearch.tsx:81]: selecting something will only give us its value; we'll need to match it to our dataset if we want whoever uses this to be able to determine its name

File: apps/app/src/views/routes/Settings/views/TreatmentTab.tsx [hunk 4]
```diff
@@ -24,7 +24,7 @@ import { ADD_TO_CATALOG } from '../../../../mutations';
 import { SplitLayout } from '../../../components/SplitLayout';
```
Comment (billoran8) [inline apps/app/src/views/routes/Settings/views/TreatmentTab.tsx:24]: this file is responsible for rendering treatments in the catalog, not all treatments.

**Comment by jknapp25** [comment]
I reckon the bolded search string will be useful here too. There's a function called `boldSubstring` that should be helpful.

image

**Comment by billoran8** [comment]
ooh such a great idea with the search terms bolding - let me integrate that

**Comment by billoran8** [comment]
i've added the nice bolding logic knapp was talking about; here's a screenshot in mobile form
Screenshot 2024-09-27 at 10 27 12 AM

File: apps/app/src/utils.ts [hunk 5]
```diff
@@ -66,6 +66,41 @@ const unique = (array: any[], propertyName: string) => {
...
 };
 
+/**
+ * This helper function will take in a provided input string and a substringToMatchOn
+ * and will return a list of parts of the input string and whether or not they match
+ * on any part of the substrcingToMatchOn
+ *
+ * This is useful for bolding parts of a search string that uses spaces.
+ */
+const getMatchingPartsFromSubstring = (
```
Comment (billoran8) [inline apps/app/src/utils.ts:76]: the corresponding logic exists in elements which i feel like is a faux pas to add here, in the react app - so i've copied the logic and have it return a list of objects that can be mapped over to determine if the parts match any part of the search string or not.

**Comment by captDaylight** [comment]
@billoran8 I don't see any updates to the `src/gql` folder, when you add a new mutation it should build the types there

File: apps/app/src/views/components/TreatmentOptionSearch.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+import { useDebounce } from 'use-debounce';
+import { SelectField } from './SelectField';
+import { getMatchingPartsFromSubstring } from '../../utils';
+
+const SearchTreatmentOptionsQuery = gql`
+  query SearchTreatmentOptions($searchTerm: String!) {
+    treatmentOptions(searchTerm: $searchTerm) {
+      id: medicationId
+      form
+      name
```
Comment (captDaylight) [inline apps/app/src/views/components/TreatmentOptionSearch.tsx:24]: We aren't using all of this info that's being fetched

File: apps/app/src/views/components/TreatmentOptionSearch.tsx (hunk 2 above)
Comment (captDaylight) [reply apps/app/src/views/components/TreatmentOptionSearch.tsx:84]: naw, this is current state of the art

File: apps/app/src/views/components/TreatmentOptionSearch.tsx (hunk 3 above)
Comment (captDaylight) [reply apps/app/src/views/components/TreatmentOptionSearch.tsx:123]: I'm not sure what you mean, you query comes back with id and name

**Review by captDaylight — APPROVED** [review]

File: apps/app/src/views/components/TreatmentOptionSearch.tsx (hunk 6 above)
Comment (billoran8) [reply apps/app/src/views/components/TreatmentOptionSearch.tsx:24]: oho snap good point

File: apps/app/apps/app/src/gql/fragment-masking.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import { Incremental } from './graphql';
```
Comment (billoran8) [inline apps/app/apps/app/src/gql/fragment-masking.ts:1]: this is weird; not sure why theyr'e new and in this diff directory; investigating.

**Pull request merged**